### PR TITLE
Make enabled metrics configurable

### LINKED ISSUE #4982: container_runtime_crio_image_* metrics may have excessive cardinality
**Description**

Since #2912, CRI-O exposes metrics about image pulls and layers:
* container_runtime_crio_image_pulls_by_digest
* container_runtime_crio_image_layer_reuse
* container_runtime_crio_image_pulls_by_name
* container_runtime_crio_image_pulls_successes

These metrics can have very high cardinality, especially because of labels such as `digest`, `size` and `name`. For some of the OpenShift CI clusters, we've seen that these metrics account for more than 50% of all metrics collected by Prometheus (e.g. about 5M series out of 10M).

I can understand how the granularity is useful for troubleshooting and debugging but labels with such cardinality can be problematic since it increases the memory usage of Prometheus (as stated in the [Prometheus docs](https://prometheus.io/docs/practices/instrumentation/#do-not-overuse-labels)). I would rather see histogram metrics that allow to view the distribution of values (e.g. `container_runtime_crio_image_pulls_bucket`).

**Steps to reproduce the issue:**
1. Run many container images with different digest IDs and many layers 
2.
3.

**Describe the results you received:**

The number of metrics exposed by CRI-O increases by a lot. 

**Describe the results you expected:**

No to little increase in the number of metrics collected.

**Additional information you deem important (e.g. issue happens only occasionally):**

We've decided for now that the cluster monitoring operator will configure Prometheus to [drop](https://github.com/openshift/cluster-monitoring-operator/blob/3c0353e0c3049ec755c9aff0aeb0b091f6d7f0a9/assets/control-plane/service-monitor-kubelet.yaml#L96-L100) these metrics to avoid excessive memory usage.

**Output of `crio --version`:**

```
1.21.1-3.rhaos4.8.gite1bc46b.el8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

AWS but valid for other environments too.

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now add a new configuration option as well as CLI flag to make the
metrics configurable. Per default all metrics are enabled, whereas it is
possible to omit or prefix the metrics collectors with
`container_runtime_` (the metrics namespace) and/or `crio_` (our own
chosen prefix).
#### Which issue(s) this PR fixes:

Fixes #4982
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added `-metrics-collectors`/`metrics_collectors` configuration to enable or disable certain metrics.
All metrics collectors are enabled per default.
```

### CODE REVIEW

File: server/metrics/collectors/collectors.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+// Collector specifies a single metrics collector identifier.
+type Collector string
+
+// Collectors specifies a list of metrics collectors.
+type Collectors []Collector
+
+const (
+	crioPrefix      = "crio_"
+	subsystemPrefix = Subsystem + "_"
```
Comment (haircommander) [inline server/metrics/collectors/collectors.go:13]: nit: can we define this below Subsystem?

Comment (saschagrunert) [reply server/metrics/collectors/collectors.go:13]: Yep, I'll have to rebase in any case if https://github.com/cri-o/cri-o/pull/5059 merges.

Comment (saschagrunert) [reply server/metrics/collectors/collectors.go]: Fixed

**Comment by saschagrunert** [comment]
Rebased, PTAL again @haircommander :)

**Comment by saschagrunert** [comment]
@haircommander friendly ping to take a look here :)

**Comment by TomSweeneyRedHat** [comment]
I was expecting to see at least a man page update?  Is it not needed?

**Comment by saschagrunert** [comment]
Two follow ups on this one to make it work in OpenShift: 
- https://github.com/openshift/cluster-monitoring-operator/pull/1287
- https://github.com/openshift/machine-config-operator/pull/2678

**Pull request merged**